File: frontend/bff/src/application/rules/commands/update-rule.command.ts
import type { RequestContext } from '../../../cross-cutting/context/request-context.js';

/** Renomeia a alçada e troca as roles concedidas (vale na hora para quem a tem). */
export interface UpdateRuleCommand {
  context: RequestContext;
  id: string;
  name: string;
  roleCodes: string[];
}

File: backend/src/Fix.Application/UseCases/Rules/Commands/UpdateRuleCommand.cs
using Fix.Application.Abstractions.Authorization;
using Fix.Application.Abstractions.Context;
using Fix.Application.Abstractions.Messaging;
using Fix.Application.Rules.Dtos;

namespace Fix.Application.Rules.Commands;

/// <summary>Renomeia a alçada e troca as roles que ela concede (vale na hora para todos que a têm).</summary>
[RequireRole(global::Fix.Domain.AggregateRoots.Roles.RoleCodes.EditOrganization)]
public sealed record UpdateRuleCommand(Guid UserId, Guid OrganizationId, Guid Id, string Name, IReadOnlyList<string> RoleCodes)
    : ICommand<RuleDto>, IOrganizationRequest;
